src/main/java/bio/terra/service/iam/sam/SamIam.java

```diff
@@ -133,6 +142,7 @@ private boolean isAuthorizedInner(AuthenticatedUserRequest userReq,
         try (Stream<ResourceAndAccessPolicy> resultStream =
                  samResourceApi.listResourcesAndPolicies(iamResourceType.toString()).stream()) {
             return resultStream
+                .filter(resource -> ValidationUtils.isValidUUID(resource.getResourceId()))
```

Comment:
Not sure I follow?  The main point of the PR is to specifically not have an exception thrown

Reply:
I think this is the important thing that needs to be commented in the code. The point is that all resources TDR creates are UUIDs, so any resources that are found that aren't UUIDs can be removed, since they don't refer to TDR resources. Otherwise it looks like we're discarding user data without logging/etc.